src/spikeinterface/core/operatorrecordings.py

```diff
@@ -0,0 +1,80 @@
+from spikeinterface.core import BaseRecording, BaseRecordingSegment
+from spikeinterface.core.recording_tools import get_rec_attributes, do_recording_attributes_match
+
+
+class BaseOperatorRecording(BaseRecording):
+    """Base class for operator recordings."""
+
+    def __init__(self, recording1, recording2, operator: str):
+        assert operator in ["add", "subtract"], "Operator must be 'add' or 'subtract'"
+        assert all(
+            isinstance(rec, BaseRecording) for rec in [recording1, recording2]
+        ), "'recordings' must be a list of RecordingExtractor"
+
+        rec_attrs2 = get_rec_attributes(recording2)
+        assert do_recording_attributes_match(
+            recording1, rec_attrs2
+        ), "Both recordings must have the same sampling frequency and channel ids"
+        assert self.check_times_kwargs(recording1, recording2), "Both recordings must have the same time parameters"
+
+        channel_ids = recording1.channel_ids
+        sampling_frequency = recording1.sampling_frequency
+        dtype = recording1.get_dtype()
+
+        BaseRecording.__init__(self, sampling_frequency, channel_ids, dtype)
+
+        for segment1, segment2 in zip(recording1._recording_segments, recording2._recording_segments):
+            add_segment = OperatorRecordingSegment(segment1, segment2, operator)
+            self.add_recording_segment(add_segment)
+
+        self._kwargs = dict(recording1=recording1, recording2=recording2, operator=operator)
+
+    def check_times_kwargs(self, recording1, recording2) -> bool:
```

Comment:
```suggestion
    def are_time_kwargs_compatible(self, recording1, recording2) -> bool:
```

More explicit and usual convention for functions that return bool.